src/server/bitops_family.cc

```diff
@@ -0,0 +1,267 @@
+#include "server/bitops_family.h"
+
+#include <bitset>
+
+#include "server/string_family.h"
+extern "C" {
+#include "redis/object.h"
+}
+
+#include "base/logging.h"
+#include "server/command_registry.h"
+#include "server/common.h"
+#include "server/conn_context.h"
+#include "server/engine_shard_set.h"
+#include "server/error.h"
+#include "server/tiered_storage.h"
+#include "server/transaction.h"
+#include "util/varz.h"
+
+namespace dfly {
+using namespace facade;
+
+namespace {
+static const int32_t OFFSET_FACTOR = 8;  // number of bits in byte
+static const long INDEX_MAPPING[] = {    // or same reason at REDIS the indenes at left to right
+    7, 6, 5, 4, 3, 2, 1, 0};
+// the offset is in bites
+// The following is the list of the functions that would handle the
+// commands that handle the bit operations
+void BitPos(CmdArgList args, ConnectionContext* cntx);
+void BitCount(CmdArgList args, ConnectionContext* cntx);
+void BitField(CmdArgList args, ConnectionContext* cntx);
+void BitFieldRo(CmdArgList args, ConnectionContext* cntx);
+void BitOp(CmdArgList args, ConnectionContext* cntx);
+void GetBit(CmdArgList args, ConnectionContext* cntx);
+void SetBit(CmdArgList args, ConnectionContext* cntx);
+OpResult<std::string> ReadValue(const OpArgs& op_args, std::string_view key);
+std::string GetString(EngineShard* shard, const PrimeValue& pv);
+// ------------------------------------------------------------------------- //
+//  Bits manipulation functions
+constexpr int32_t GetBitIndex(int32_t offset) noexcept {
+  return offset % OFFSET_FACTOR;
+}
+
+constexpr int32_t GetByteIndex(int32_t offset) noexcept {
+  return offset / OFFSET_FACTOR;
+}
+
+std::string::value_type GetByteValue(const std::string& value, int32_t offset) {
+  return value[GetByteIndex(offset)];
+}
+
+long GetBitValue(const std::string& value, int32_t offset) {
```

Comment:
is there a reason for choosing a return type `long` and choosing `offset` as int32_t and not uint32_t or size_t ?

Reply:
regarding return type, I meant long is a bit wide for bit, maybe use `int` or even `uint8_t` ?

Reply:
we are returning in "SendLong" a long value, and the Redis doc says that it would be in the range of int 32. This is the only reason. I can change that

Reply:
it's true that redis protocol only supports "long" but there is no reason to handle it internally in a more convenient way. 
Functions like `SetBit` and `GetBit` are intended to separate the internal implementation from the protocol and redis command structure.

Reply:
I changed all the values for the bits to be bool for both setting a bit and return the value of the bit, In the "API" for the command it would return long to be consistent with redis